### PR TITLE
Log vehicle local position setpoints for fixedwing position control

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
**Describe problem solved by this pull request**
Previously it was hard to interpret what the fw position controller was doing in term of tracking performance since the position setpoints were not logged in the local frame.

- It is hard to associate waypoints switching with the vehicle behavior, given that the fw pos control overrides the setpoints internally sometimes
- This is especially a problem when passing high bandwidth setpoints from offboard or for path tracking.
- For VTOLs the position setpoints just stops logging when the vehicle transitions to FW mode, since the local_position_setpoint is only published from the MC position controller module

**Describe your solution**
This adds publication of the `vehicle_local_position_setpoint` so that it gets logged as a local setpoint.

**Describe possible alternatives**
- I think a bigger problem creeping is that all our position setpoints are in the global frame due to L1 only supporting lat/lon as a input. For offboard mode, we are converting local->global and global->local two times for this reason. I would like to move towards doing fw control in local coordinates since this would allow us to fly even in GPS denied situations. 

**Test data / coverage**
Tested in SITL Gazebo
```
make px4_sitl gazebo_plane
```
and the log correctly shows the position setpoints and how the waypoints are switching.
- Before PR: [FlightLog](https://logs.px4.io/plot_app?log=290f3041-74cf-4126-9b63-d3eeea42f058)
image

- After PR: [Flightlog](https://logs.px4.io/plot_app?log=75f1c7b7-bbf7-4228-954c-67f3c35fb1b9)
image

**Additional context**

-

### CODE REVIEW

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 1]
```diff
@@ -2237,6 +2238,22 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const hrt_abstime &now, flo
 	tecs_status_publish();
 }
 
+void FixedwingPositionControl::publishLocalPositionSetpoint(const position_setpoint_s current_waypoint)
```
Comment (dagar) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:2241]:
```suggestion
void FixedwingPositionControl::publishLocalPositionSetpoint(const position_setpoint_s& current_waypoint)
```

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 2]
```diff
@@ -2237,6 +2238,22 @@ FixedwingPositionControl::tecs_update_pitch_throttle(const hrt_abstime &now, flo
...
+{
+	if (_global_local_proj_ref.isInitialized()) {
+		Vector2f current_setpoint = _global_local_proj_ref.project(current_waypoint.lat, current_waypoint.lon);
+		vehicle_local_position_setpoint_s local_position_setpoint{};
+		local_position_setpoint.x = current_setpoint(0);
```
Comment (dagar) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:2246]: populate the unused fields with NANs?

`local_position_setpoint.vx = NAN`, etc

Comment (dagar) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:2246]: Alternatively does populating any other existing data make sense (for casual interest)?
 - acceleration from L1?
 - velocities from TECS (remove wind?)

**Comment by dagar** [comment]
Looks good, although doesn't this need to be skipped during VTOL transitions?

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 2 above)
Comment (Jaeyoung-Lim) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:2249]: I think for now IMO there is not much more to add - e.g. velocity setpoints are not velocity but airspeed setpoints / accelerations are not actual acceleration setpoints but roll setpoints. I think we can add these by cleaning up the controller one by one

**Comment by Jaeyoung-Lim** [comment]
> Looks good, although doesn't this need to be skipped during VTOL transitions?

Done! 

Happy new year! :tada:

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 1 above)
Comment (Jaeyoung-Lim) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp]: Thanks, Done!

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp (hunk 2 above)
Comment (sfuhrer) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp:2249]: Some stuff - e.g. lateral acceleration- will be published with the new L1 controller (https://github.com/PX4/PX4-Autopilot/pull/18810) in npfg_status. I would also find it nicer if we publish whatever we can in local_position_setpoint. That though also requires some controller interface alignment - using local coordinates prob is one thing of that.

**Review by sfuhrer — APPROVED** [review]
Thanks, that makes sense.

Let's figure out if it makes sense to pull some more data out of tecs_status and npfg_status into vehicle_local_position setpoint.

**Comment by Jaeyoung-Lim** [comment]
> Some stuff - e.g. lateral acceleration- will be published with the new L1 controller (#18810) in npfg_status. I would also find it nicer if we publish whatever we can in local_position_setpoint. That though also requires some controller interface alignment - using local coordinates prob is one thing of that.

Fully agreed! I think now that we have two controllers we can use this as an opportunity to clean up the controller interface

**Pull request merged**